datafusion/src/execution/context.rs

```diff
@@ -1148,8 +1180,6 @@ pub struct ExecutionContextState {
     pub catalog_list: Arc<dyn CatalogList>,
     /// Scalar functions that are registered with the context
     pub scalar_functions: HashMap<String, Arc<ScalarUDF>>,
-    /// Variable provider that are registered with the context
-    pub var_provider: HashMap<VarType, Arc<dyn VarProvider + Send + Sync>>,
```

Comment:
while technically this is a breaking change, the  `ExecutionContext::register_variable` remains the same which I think means this change will be minimally disruptive to people using DataFusion